panic when removing subdomain on domain with multiple subdomains

```
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/threefoldtech/tfgateway/dns.(*Zone).Remove(0xc000169048, 0x15413a0, 0xc0004507e0)
        /home/Jo/code/gopath/src/github.com/threefoldtech/tfgateway/dns/dns_zone.go:79 +0x4e1
github.com/threefoldtech/tfgateway/dns.(*Mgr).RemoveSubdomain(0xc000429fc0, 0xc00037f458, 0x3, 0xc0004b42e0, 0x16, 0xc0003aae40, 0x
        /home/Jo/code/gopath/src/github.com/threefoldtech/tfgateway/dns/coredns.go:237 +0x3b4
github.com/threefoldtech/tfgateway.(*Provisioner).subDomainDecomission(0xc0000f50b0, 0x15637a0, 0xc0003fd000, 0xc00040fb80, 0xc0003
        /home/Jo/code/gopath/src/github.com/threefoldtech/tfgateway/subdomain.go:37 +0x20c
github.com/threefoldtech/zos/pkg/provision.(*Engine).decommission(0xc000169bf0, 0x15637a0, 0xc0003fd000, 0xc00040fb80, 0x0, 0xc0002
        /home/Jo/code/gopath/pkg/mod/github.com/threefoldtech/zos@v0.2.6/pkg/provision/engine.go:196 +0x109
github.com/threefoldtech/zos/pkg/provision.(*Engine).Run(0xc000169bf0, 0x15637a0, 0xc0003fd000, 0x0, 0x0)
        /home/Jo/code/gopath/pkg/mod/github.com/threefoldtech/zos@v0.2.6/pkg/provision/engine.go:110 +0x7e4
main.run(0xc000299cc0, 0x0, 0x0)
        /home/Jo/code/gopath/src/github.com/threefoldtech/tfgateway/cmd/tfgateway/main.go:222 +0x15a5
github.com/urfave/cli/v2.(*App).RunContext(0x1b895e0, 0x15637e0, 0xc00003c0a0, 0xc0000320b0, 0xb, 0xb, 0x0, 0x0)
        /home/Jo/code/gopath/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:315 +0x70b
github.com/urfave/cli/v2.(*App).Run(...)
        /home/Jo/code/gopath/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
main.main()
        /home/Jo/code/gopath/src/github.com/threefoldtech/tfgateway/cmd/tfgateway/main.go:98 +0x6d
```

--- a/dns/dns_zone.go
+++ b/dns/dns_zone.go
@@ -70,16 +70,18 @@ func (z *Zone) Remove(r Record) {
 		z.Records = records{}
 	}
 
-	_, ok := z.Records[r.Type()]
+	records, ok := z.Records[r.Type()]
 	if !ok {
 		return
 	}
 
-	for i := range z.Records[r.Type()] {
-		if z.Records[r.Type()][i] == r {
-			z.Records[r.Type()] = append(z.Records[r.Type()][:i], z.Records[r.Type()][i+1:]...)
+	newrecords := records[:0]
+	for _, record := range records {
+		if record != r {
+			newrecords = append(newrecords, record)
 		}
 	}
+	z.Records[r.Type()] = newrecords
 }
 
 type records map[RecordType][]Record

--- a/dns/dns_zone_test.go
+++ b/dns/dns_zone_test.go
@@ -13,6 +13,14 @@ func TestZone(t *testing.T) {
 		IP4: "10.0.0.1",
 		TTL: 3600,
 	}
+	b := RecordA{
+		IP4: "10.0.0.2",
+		TTL: 3600,
+	}
+	c := RecordA{
+		IP4: "10.0.0.3",
+		TTL: 3600,
+	}
 	aaaa := RecordAAAA{
 		IP6: "fdb5:7faa:7be7:3ef3::1",
 		TTL: 3600,
@@ -34,4 +42,11 @@ func TestZone(t *testing.T) {
 	// remove no existing also works
 	z.Remove(aaaa)
 	assert.Equal(t, 0, len(z.Records[RecordTypeAAAA]))
+
+	// add 2 more and remove one in the middle
+	z.Add(b)
+	z.Add(c)
+	assert.Equal(t, 3, len(z.Records[RecordTypeA]))
+	z.Remove(b)
+	assert.Equal(t, 2, len(z.Records[RecordTypeA]))
 }

